fairseq/trainer.py

```diff
@@ -344,7 +345,12 @@ def maybe_no_sync():
             self._prev_grad_norm = grad_norm
 
             # take an optimization step
-            # self.optimizer.step()  # XLA: this is done by torch_xla
+
+            # xla takes care of optimization step using
+            #   torch_xla.xla_model.optimizer_step
+            # so skip optimization step here in that case
+            if not self.xla:
```

Comment:
What if you let a lambda in, which defaults to optimizer.step(), which we override to an empty function?

Reply:
I can do that, but I thought I could use this flag for other types of behavior elsewhere in the object in the future. Would you like me to switch to lambda?

Reply:
Up to you. I only have a limited view of the overall model.

Reply:
ok let's go with this then, I can switch to lambda any time we want.